src/detect-engine-siggroup.c

```diff
@@ -779,19 +779,15 @@ int SigAddressPrepareStage1(DetectEngineCtx *);
  */
 static int SigGroupHeadTest03(void)
 {
-    int result = 1;
-
     DetectEngineCtx de_ctx;
 
     SigGroupHeadHashInit(&de_ctx);
-
-    result &= (de_ctx.sgh_hash_table != NULL);
+    FAIL_IF_NOT(de_ctx.sgh_hash_table);
```

Comment:
Although this is correct, I think it'd be clearer if the macros `FAIL_IF_NULL` and `FAIL_IF_NOT_NULL` were used.

Reply:
Indeed, Victor has also given that feedback in some of my PRs: it is preferred to use the macros with explicit `NULL`, when dealing with pointers.